lib/model/folder.go

```diff
@@ -262,14 +271,231 @@ func (f *folder) getHealthError() error {
 }
 
 func (f *folder) scanSubdirs(subDirs []string) error {
-	if err := f.model.internalScanFolderSubdirs(f.ctx, f.folderID, subDirs, f.localFlags); err != nil {
-		// Potentially sets the error twice, once in the scanner just
-		// by doing a check, and once here, if the error returned is
-		// the same one as returned by CheckHealth, though
-		// duplicate set is handled by setError.
+	if err := f.CheckHealth(); err != nil {
```

Comment:
This change is very confusing, I guess that's just unwinding a single stack level? Or is it doing more as the diff is huge, hence it's hard to tell.

Reply:
It moves the code from `Model.internalScanFolderSubdirs` (https://github.com/syncthing/syncthing/pull/5215/files#diff-1df9805b2381b6288b220226f82d5dc9L1968) into `folder.scanSubdirs`. There are no functional changes, but a few things had to be adjusted to the new underlying type. I could rebase the PR making a first commit, where I just copy the function over (won't build) and a second commit, where I make it work - would that help?